icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitsData.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *******************************************************************************
+ * Copyright (C) 2004-2020, Google Inc, International Business Machines
+ * Corporation and others. All Rights Reserved.
+ *******************************************************************************
+ */
+
+package com.ibm.icu.impl.units;
+
+import com.ibm.icu.impl.Assert;
+import com.ibm.icu.impl.ICUData;
+import com.ibm.icu.impl.ICUResourceBundle;
+import com.ibm.icu.impl.UResource;
+import com.ibm.icu.util.UResourceBundle;
+
+import java.util.ArrayList;
+
+/**
+ * Responsible for all units data operations (retriever, analysis, extraction certain data ... etc.).
+ */
+public class UnitsData {
+    private static String[] simpleUnits = null;
+
+    public static String[] getSimpleUnits() {
+        if (simpleUnits != null) {
+            return simpleUnits;
+        }
+
+        // Read simple units
+        ICUResourceBundle resource;
+        resource = (ICUResourceBundle) UResourceBundle.getBundleInstance(ICUData.ICU_BASE_NAME, "units");
+        SimpleUnitsSink sink = new SimpleUnitsSink();
+        resource.getAllItemsWithFallback("convertUnits", sink);
+        simpleUnits = sink.simpleUnits;
+
+        return simpleUnits;
+    }
+
+    public static class SimpleUnitsSink extends UResource.Sink {
+        String[] simpleUnits = null;
+
+        @Override
+        public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
+            Assert.assrt(key.toString().equals(Constants.CONVERSION_UNIT_TABLE_NAME));
+            Assert.assrt(value.getType() == UResourceBundle.TABLE);
+
+            UResource.Table simpleUnitsTable = value.getTable();
+            ArrayList<String> simpleUnits = new ArrayList<>();
+            for (int i = 0; simpleUnitsTable.getKeyAndValue(i, key, value); i++) {
+                simpleUnits.add(key.toString());
```

Comment:
Also include the kilogram fix in #1257 

CC @hugovdm

Reply:
I have added it